site/src/components/ErrorSummary/ErrorSummary.tsx

```diff
@@ -1,32 +1,112 @@
+import Link from "@material-ui/core/Link"
+import { makeStyles, Theme } from "@material-ui/core/styles"
+import CloseIcon from "@material-ui/icons/Close"
 import RefreshIcon from "@material-ui/icons/Refresh"
-import { FC } from "react"
+import { ApiError, getErrorDetail, getErrorMessage } from "../../api/errors"
+import { FC, useState } from "react"
 import { Stack } from "../Stack/Stack"
 
 const Language = {
   retryMessage: "Retry",
   unknownErrorMessage: "An unknown error has occurred",
+  moreDetails: "More",
+  lessDetails: "Less",
 }
 
 export interface ErrorSummaryProps {
-  error: Error | unknown
+  error: ApiError | Error | unknown
   retry?: () => void
+  dismissible?: boolean
+  defaultMessage?: string
 }
 
-export const ErrorSummary: FC<ErrorSummaryProps> = ({ error, retry }) => (
-  <Stack>
-    {!(error instanceof Error) ? (
-      <div>{Language.unknownErrorMessage}</div>
-    ) : (
-      <div>{error.toString()}</div>
-    )}
-
-    {retry && (
-      <div>
-        <Button onClick={retry} startIcon={<RefreshIcon />} variant="outlined">
-          {Language.retryMessage}
-        </Button>
-      </div>
-    )}
-  </Stack>
-)
+export const ErrorSummary: FC<ErrorSummaryProps> = ({
+  error,
+  retry,
+  dismissible,
+  defaultMessage,
+}) => {
+  const message = getErrorMessage(error, defaultMessage || Language.unknownErrorMessage)
+  const detail = getErrorDetail(error)
+  const [showDetails, setShowDetails] = useState(false)
+  const [isOpen, setOpen] = useState(true)
+
+  const styles = useStyles({ showDetails })
+
+  const toggleShowDetails = () => {
+    setShowDetails(!showDetails)
+  }
+
+  const closeError = () => {
+    setOpen(false)
+  }
+
+  if (!isOpen) {
+    return null
+  }
+
+  return (
+    <Stack>
+      <Stack className={styles.root}>
+        <Stack direction="row" alignItems="center" className={styles.message}>
+          <div>
+            <span className={styles.errorMessage}>{message}</span>
+            {!!detail && (
+              <Link onClick={toggleShowDetails} className={styles.detailsLink}>
+                {showDetails ? Language.lessDetails : Language.moreDetails}
+              </Link>
+            )}
+          </div>
+          {dismissible && (
+            <IconButton onClick={closeError} className={styles.iconButton}>
+              <CloseIcon className={styles.closeIcon} />
+            </IconButton>
+          )}
+        </Stack>
+        <Collapse in={showDetails}>{detail}</Collapse>
+      </Stack>
+
+      {retry && (
+        <div>
+          <Button onClick={retry} startIcon={<RefreshIcon />} variant="outlined">
+            {Language.retryMessage}
+          </Button>
+        </div>
+      )}
+    </Stack>
+  )
+}
+
+interface StyleProps {
+  showDetails?: boolean
+}
+
+const useStyles = makeStyles<Theme, StyleProps>((theme) => ({
+  root: {
+    background: `${theme.palette.error.main}60`,
+    margin: `${theme.spacing(2)}px`,
```

Comment:
Just a suggestion: this component might be a little more flexible if we leave off the margin and instead let the parent decide what the margin should be (it will probably vary). If you feel like it, you could even pass in a `style` prop so that the parent can pass through its own styles.

Reply:
I think we're going to use this component a lot and I'd like it to look decent without additional styling. Maybe we can give it margins we think are typical and then override when necessary.